test/c3/appframework/appframework_test.go

```diff
@@ -100,79 +115,280 @@ var _ = Describe("c3appfw test", func() {
+			   * Prepare C3 CRD with MC and App config
+			   * Apply C3 CRD and wait for pod to be ready
+			   * Prepare MC CRD with App Config
+			   * Wait for MC to be READY
+			   ################## VERIFICATIONS #############
+			   * Verify apps are copied, installed on MC AND SH,Indexers pods
+			   ##########  SCALE UP SHC and Indexers ###########
+			   * Wait for SHC, indexers and MC to be ready
+			   * Verify apps are copied, installed on MC AND SH,Indexers pods
+			   ############ DOWNGRADE APPS #############################
+			   * Downgrade apps in app sources
+			   * Wait for MC and C3 to be READY
+			   * Verify apps are copied, installed and downgraded on MC AND SH,Indexers pods
+			*/
+
+			// Delete pre-installed apps on S3
+			testenv.DeleteFilesOnS3(testS3Bucket, uploadedApps)
+			uploadedApps = nil
+			testenvInstance.Log.Info("Testing downgrade scenario")
+
+			// Upload newer version of apps to S3
+			s3TestDir = "c3appfw-" + testenv.RandomDNSName(4)
+			appFileList := testenv.GetAppFileList(appListV2, 2)
+			appVersion := "V2"
+			uploadedFiles, err := testenv.UploadFilesToS3(testS3Bucket, s3TestDir, appFileList, downloadDirV2)
+			Expect(err).To(Succeed(), "Unable to upload apps to S3 test directory")
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+			s3TestDirMC := "c3appfw-mc-" + testenv.RandomDNSName(4)
+			uploadedFiles, err = testenv.UploadFilesToS3(testS3Bucket, s3TestDirMC, appFileList, downloadDirV2)
+			Expect(err).To(Succeed(), "Unable to upload apps to S3 test directory")
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+
+			// Create App framework Spec for C3
+			volumeName := "appframework-test-volume-" + testenv.RandomDNSName(3)
+			volumeSpec := []enterpriseApi.VolumeSpec{testenv.GenerateIndexVolumeSpec(volumeName, testenv.GetS3Endpoint(), testenvInstance.GetIndexSecretName(), "aws", "s3")}
+
+			// AppSourceDefaultSpec: Remote Storage volume name and Scope of App deployment
+			appSourceDefaultSpec := enterpriseApi.AppSourceDefaultSpec{
+				VolName: volumeName,
+				Scope:   enterpriseApi.ScopeCluster,
+			}
+
+			// appSourceSpec: App source name, location and volume name and scope from appSourceDefaultSpec
+			appSourceName := "appframework" + testenv.RandomDNSName(3)
+			appSourceSpec := []enterpriseApi.AppSourceSpec{testenv.GenerateAppSourceSpec(appSourceName, s3TestDir, appSourceDefaultSpec)}
+
+			// appFrameworkSpec: AppSource settings, Poll Interval, volumes, appSources on volumes
+			appFrameworkSpec := enterpriseApi.AppFrameworkSpec{
+				Defaults:             appSourceDefaultSpec,
+				AppsRepoPollInterval: 60,
+				VolList:              volumeSpec,
+				AppSources:           appSourceSpec,
+			}
+
+			indexerReplicas := 3
+
+			testenvInstance.Log.Info("Deploy Single Site Indexer Cluster")
+			err = deployment.DeploySingleSiteClusterWithGivenAppFrameworkSpec(deployment.GetName(), indexerReplicas, true, appFrameworkSpec, 10)
```

Comment:
`monitoringConsoleRef` is not added to the config. The cluster will not be configured on MC itself.